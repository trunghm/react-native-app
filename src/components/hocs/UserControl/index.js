import React, {Component, useState} from 'react';
import PropTypes from 'prop-types';
import styles from "./styles";
import {
  StyleSheet,
  Text,
  View,
  TouchableOpacity
} from 'react-native';
import {GroupButton} from "../../modules";
import {Avatar} from "../../commons";

export default function UserControl({active}) {
  const [_active, setActive] = useState(active);

  return (
    <View style={styles().container}>
      <Avatar text={"H"}/>
      <GroupButton index={_active ? 0 : 1} data={["Deactive User", "Active User"]} onSelect={(index) => {
        setActive(index === 0 ? true : false);
      }}/>
    </View>);
}


