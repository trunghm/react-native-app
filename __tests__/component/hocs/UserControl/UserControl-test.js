/**
 * @format
 */

import 'react-native';
import React from 'react';
import UserControl from '../../../../src/components/hocs/UserControl';

// Note: test renderer must be required after react-native.
import {mount, shallow} from 'enzyme';
import renderer from  "react-test-renderer";
import EStyleSheet from 'react-native-extended-stylesheet';


const originalError = console.error;

beforeAll(() => {
  console.error = jest.fn();
  EStyleSheet.build({
  });
});

afterAll(() => {
  console.error = originalError;
});

describe('Testing  UserControl component', () => {

  it('renders snapshot as expected', () => {
    const wrapper = renderer.create(
      <UserControl active={true}/>
    );

    expect(wrapper).toMatchSnapshot();

  });

  it('renders as expected', () => {
    const wrapper = mount(
      <UserControl active={true}/>
    );

    expect(wrapper).toMatchSnapshot();
    expect( wrapper
      .findWhere((w) => w.prop('text') === 'Deactive User')
      .first().prop('disabled') ).toEqual(false);
    expect( wrapper
      .findWhere((w) => w.prop('text') === 'Active User')
      .first().prop('disabled') ).toEqual(true);

  });

});
