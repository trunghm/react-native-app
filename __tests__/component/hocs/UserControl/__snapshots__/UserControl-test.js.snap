// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing  UserControl component renders as expected 1`] = `ReactWrapper {}`;

exports[`Testing  UserControl component renders snapshot as expected 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#d9e2e7",
        "borderRadius": 45,
        "borderWidth": 0.5,
        "height": 45,
        "justifyContent": "center",
        "width": 45,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1496bb",
        }
      }
    >
      H
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "maxHeight": 50,
        "width": "100%",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#1496bb",
          "flex": 1,
          "justifyContent": "center",
          "maxHeight": 50,
          "minHeight": 50,
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        DEACTIVE USER
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#d2d2d2",
          "flex": 1,
          "justifyContent": "center",
          "maxHeight": 50,
          "minHeight": 50,
          "opacity": 1,
        }
      }
    >
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#FFFFFF",
            "fontSize": 14,
            "fontWeight": "bold",
          }
        }
      >
        ACTIVE USER
      </Text>
    </View>
  </View>
</View>
`;
